src/Query/Aggregations/Metric/WeightedAvg.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace AviationCode\Elasticsearch\Query\Aggregations\Metric;
+
+use AviationCode\Elasticsearch\Model\Aggregations\Metric\WeightedAvg as WeightedAvgModel;
+
+class WeightedAvg extends Metric
+{
+    /**
+     * @var array
+     */
+    private $value;
+
+    /**
+     * @var array
+     */
+    private $weight;
+
+    public function __construct(array $value, array $weight)
+    {
+        parent::__construct('weighted_avg', WeightedAvgModel::class);
+
+        $this->value = $value;
+
+        $this->weight = $weight;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function toElastic(): array
```

Comment:
I think we are missing `field` here.

The basic example they give in the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-weight-avg-aggregation.html#_examples)

```json
        "weighted_grade": {
            "weighted_avg": {
                "value": {
                    "field": "grade"
                },
                "weight": {
                    "field": "weight"
                }
            }
        }
```

We could use the weighted average as follows

```php
$weightedAvg = new WeightedAverage('weighted_grade', 'grade', 'weight');
```

Optionally be able to set the missing fields

```php
$weightedAvg = new WeightedAverage('weighted_grade', 'grade', 'weight');
$weightedAvg->setWeightMissing(2.0)->setValueMissing(3.0);
```

Reply:
@Douglasdc3 the $value and $weight class variables are of type `array`, so from outside you could do:
```
$value = ['field' => 'price', 'missing' => 10];
$weight = ['field' => 'sales_percentage', 'missing' => 0.10];
$aggs->weightedAvg('weighted_price', $value, $weight);
```
As seen above, 'field' is there.

 I think it's better of to just pass in an array type $weight, $value with the params as the user wants (i.e. could also specify `missing` option) to the constructor of weightedAvg aggregation, instead of implementing extra`setWeightMissing($missingWeight)` and `setValueMissing($missingValue)` . What do you think ?